src/uController.SourceGenerator/uControllerGenerator.cs

```diff
@@ -335,6 +335,11 @@ IOperation ResolveDeclarationOperation(ISymbol symbol)
                         }
                         populateMetadata.AppendLine($@"                PopulateMetadata<{p.ParameterType}>(parameterInfos[{p.Index}], builder);");
                     }
+
+                    if (p.BodyOrService && generatedBodyOrService || p.FromBody)
```

Comment:
This isn't right. This needs to generate conditional code if we're in the BodyOrService case.

```
if (p.FromBody)
{
    // Generate the code
}
else if (p.BodyOrService)
{
   // Generate code at that will check for if the parameter was not a service
   // and only then, add the metadata.
}
```

Also, add a test that verifies the service parameters do not have this metadata (with your change it will).